### PR TITLE
Refactored MsbContext.

### PR DESCRIPTION
**Pull request by YuriySavchuk**

### CODE REVIEW

File: core/src/test/java/io/github/tcdl/RequesterTest.java [hunk 1]
```diff
@@ -152,7 +152,8 @@ private Requester initRequesterForResponsesWithTimeout(int numberOfResponses) th
         when(channelManagerMock.findOrCreateProducer(anyString())).thenReturn(producerMock);
 
         MsbContext msbContext = TestUtils.createSimpleMsbContext();
-        msbContext.setChannelManager(channelManagerMock);
+        msbContext = TestUtils.createMsbContext(msbContext.getMsbConfig(), msbContext.getMessageFactory(),
```
Comment (vso-tc) [inline core/src/test/java/io/github/tcdl/RequesterTest.java]: Looks tricky. Couldn't we provide an easier way to specify channel manager instance?

File: core/src/test/java/io/github/tcdl/support/TestUtils.java [hunk 2]
```diff
@@ -145,5 +150,13 @@ public static Payload createSimpleResponsePayload() {
     public static MetaMessageBuilder createSimpleMetaBuilder(MsbConfigurations msbConf, Clock clock) {
         return new MetaMessage.MetaMessageBuilder(null, clock.instant(), msbConf.getServiceDetails(), clock);
     }
+    
+    private static class TestMsbContext extends MsbContext {
```
Comment (vso-tc) [inline core/src/test/java/io/github/tcdl/support/TestUtils.java]: With such trivial subclass couldn't we use MsbContext itself?

Comment (YuriySavchuk) [inline core/src/test/java/io/github/tcdl/support/TestUtils.java]: Now MsbContext has only protected Constructor. And new instance can be created only by means of MsbContext.Bulder (as a usual way) or with extended class (for test purposes).

File: core/src/test/java/io/github/tcdl/RequesterTest.java (hunk 1 above)
Comment (YuriySavchuk) [inline core/src/test/java/io/github/tcdl/RequesterTest.java]: MsbContext couldn't provide setters accessible for the end-user.So, setters were removed from the class. Mock implementations of the aggregated objects can be added either through Constructor directly or with using Builder pattern. The simplest approach with Constructor has been implemented.

**Pull request merged**